Gems/EMotionFX/Code/EMotionFX/Tools/EMotionStudio/EMStudioSDK/Source/MainWindow.cpp

```diff
@@ -2813,6 +2818,55 @@ namespace EMStudio
         }
     }
 
-} // namespace EMStudio
+    void MainWindow::UpdatePlugins(float timeDelta)
+    {
+        EMStudio::PluginManager* pluginManager = EMStudio::GetPluginManager();
+        if (!pluginManager)
+        {
+            return;
+        }
+
+        const size_t numPlugins = pluginManager->GetNumActivePlugins();
+        for (size_t i = 0; i < numPlugins; ++i)
+        {
+            EMStudio::EMStudioPlugin* plugin = pluginManager->GetActivePlugin(i);
+            plugin->ProcessFrame(timeDelta);
+        }
+    }
 
-#include <EMotionFX/Tools/EMotionStudio/EMStudioSDK/Source/moc_MainWindow.cpp>
+    void MainWindow::EnableUpdatingPlugins()
+    {
+        AZ::TickBus::Handler::BusConnect();
+    }
+
+    void MainWindow::DisableUpdatingPlugins()
+    {
+        AZ::TickBus::Handler::BusDisconnect();
+    }
+
+    void MainWindow::OnTick(float delta, AZ::ScriptTimePoint timePoint)
+    {
+        AZ_UNUSED(timePoint);
+
+        // Check if we are in game mode.
+        IEditor* editor = nullptr;
+        EBUS_EVENT_RESULT(editor, AzToolsFramework::EditorRequests::Bus, GetEditor);
+        const bool inGameMode = editor ? editor->IsInGameMode() : false;
```

Comment:
The editor doesn't change once initialized, consider saving IEditor* as a member pointer in the class and only set it once? It's not much but it reduces an ebus call each frame

Reply:
I'd like to keep the splash radius of Cry code minimal and went this way because of that. Let me know if you feel strongly about this one.

Reply:
nitpick: I think in general we prefer calling the bus directly instead of using the EBUS_EVENT_RESULT marco.
call this instead AzToolsFramework::EditorRequests::BroardcastResult

Reply:
Changed